### ISSUE TITLE
dropping pixels during trends -> incomplete trends timeseries

### ISSUE THREAD

**Issue by robinkraft** [wri/forma-clj#197]
## TL;DR

Exactly a year of trends data is missing for 2.9 million pixels, out of about 20 million, so we can't continue on to creating beta vectors just yet. Some of these pixels were dropped for producing singular matrices, but that should have been well less than 1% of periods.
## Details

The [trends cleanup query](https://github.com/reddmetrics/forma-clj/blob/9f57691f68b8f992bb5976a275635c6641b7d1a9/src/clj/forma/hadoop/jobs/forma.clj#L176-182) builds timeseries of the pixel-period level trends output. The query below It counts how many values are in the resulting timeseries. We expect 147 values, one for each period since the training period.

``` clojure
(let [src (hfs-seqfile (str
                            "s3n://formatemp/screened-trends-cleanup/"
                            "500-16"))]
      (<- [?count ?num-periods ?diff]
          (src ?s-res ?mh ?mv ?s ?l ?start ?end ?short _ _ _)
          (count ?short :> ?num-periods)
          (- ?end ?start :> ?periods)
          (inc ?periods :> ?diff)
          (c/count ?count)))
```

But as you can see in this [Google spreadsheet](https://docs.google.com/spreadsheet/ccc?key=0AiSXybs6SSQidGFRUVd5NW1VNEkwdmZFMHNEUElpeUE), fully 2.9 million have exactly 124 periods, or one year less of data than they should. Here's a selection.

``` text
count   periods of ts stats
2,922,759   124
300 125
129 126
30  127
...
17,585,291  147
```

Here are the tiles with the worst offenders:

``` text
mod-h   mod-v   # of pixels with incomplete ts
9   7   602,148
27  7   360,343
26  7   349,709
10  7   301,118
26  6   219,910
```

That table came from this query:

``` clojure
  (?- (hfs-textline "s3n://formatemp/tile-count-cleanup" :sinkmode :replace)
    (let [src (hfs-seqfile (str
                            "s3n://formatemp/screened-trends-cleanup/"
                            "500-16"))]
      (<- [?mh ?mv ?count]
          (src ?s-res ?mh ?mv ?s ?l ?start ?end ?short _ _ _)
          (count ?short :> ?short-count)
          (> 147 ?short-count)
          (c/count ?count))))
```
## Conclusions and next steps

There aren't any missing raw MODIS data - all tiles are where they should be. I could check subsequent steps in the workflow to ensure that the timeseries lengths are correct, but the takeaway for now is that we can't build the trends output timeseries that we need to get through estimation.

I've written [a query](https://github.com/reddmetrics/forma-clj/compare/reddmetrics:5b9f1f7...reddmetrics:6a7e30a) that will figure out which pixels are missing trends data and for which periods. This should help isolate the issue. It would be pretty interesting if the missing year of trends data were all for the same periods.

Worst case, we isolate these bad pixels, re-run trends just for them, and see what breaks. They can't all produce singular matrices!
